core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -403,7 +401,30 @@ class LogManager(logDirs: Seq[File],
   /**
    *  Start the background threads to flush logs and do log cleanup
    */
-  def startup(): Unit = {
+  def startup(retrieveTopicNames: => Set[String]): Unit = {
```

Comment:
Feels slightly odd to pass in the set of topics to load here, but I can't think of a good way to avoid it. Perhaps we could pass MetadataCache into LogManager and let startup call MetadataCache#getAllTopics? That might be more risky though since it changes the startup order in KafkaServer, maybe we can look into this as a follow-up.

Besides that, the name here seems strange. Maybe something like "topicsToLoad"?

Reply:
Would like to keep it as a verb: `retrieveTopicsToLoad`

Reply:
Actually, this isn't the list of topics to load -- it's the list of topics for which we will look for log config overrides.  `retrieveTopicsForLogConfigOverrideCheck`